### DISCUSSION TITLE
S100105 Pump

### DISCUSSION THREAD

**Discussion by Muhleghurts** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#663] [General]
I have an S100105 pump 6wire. I purchased a v2b pre assembled and I cannot get this act right. I’ve tried all 6 wire configurations. The 6 wire pre2021 shows the display correctly but as soon as you try and change anything it spams switches Celsius and  Fahrenheit. Is there something anyone has found to work with this pump? This is my second ever ESP project. Thanks in advance. 

![IMG_2260](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/155792181/47cdb704-0e1d-480f-9764-91fe0dd80177)

![IMG_2257](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/155792181/8e6852c9-e408-4aca-b755-3314bd097222)
![IMG_2259](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/155792181/d3a03037-1dac-4341-9e8c-b24f2237e3f9)

**Comment by metaljay** [comment]
Is this one of mine?  If yes, I tested on mine before I shipped using ‘6 wire, 2021 (Air)’ second one down (for both CIO and DSP). Select V2B and then fully turn off the hot tub. Give it a minute or two and then turn the hot tub back on. 

I know sounds silly, but cold rebooting rather than pressing the ESP reboot seems to sometimes save the config better

**Comment by visualapproach** [comment]
Looks like there is fewer than 6 wires on the cable from the pump

**Comment by colmboreilly** [comment]
I'm having the same issue, with the same pump model, and also with a new V2B board from metaljay.  It is indeed a 6 wire cable.
at first boot up it is stable, but as soon as I toggle the C/F it starts toggling repeatedly, and will not stop.  None of the other buttons seem to work correctly either.  If I toggle the heater on, the bubbler comes on instead.  I have tried multiple different HW options in the HWconfig, to no avail.
The most stable option has been the pre-2021 for CIO and DSP.  
And have also tried doing a cold restart, no good there either.
Any insight?

**Reply by Muhleghurts** [reply]
I did what @metaljay suggested and it’s been working fine ever since. You have to set your cio and dsp settings, turn off your ESP, then unplug the spa and let it sit for a solid 5 minutes before plugging it back in.

**Reply by metaljay** [reply]
Have you selected V2B board config too? It sounds like one or more of your settings aren’t saving as the fact you can toggle bubbles means commands are being sent correctly. (I.e. no damaged cables). 

You could always try reprogram it with USB if you can’t clear the settings you have entered.

**Reply by colmboreilly** [reply]
Retried again tonight, and tried with power disconnected after redoing the hardware config for 5+ minutes and it did finally work.  Where 2-3 mins didn't.   Odd behavior for sure. 

For clarity, here is the config that worked on my Coleman Spa with the square Bestway '105 pump. 

![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/7662181/f80241df-0e64-48be-8077-ebdee64b53e1)

Thank you all for the help.

**Reply by metaljay** [reply]
Ace, could you reply on my thread?
